thanos/charts/templates/ruler.yaml

```
@@ -0,0 +1,32 @@
+{{/* 
+SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+SPDX-License-Identifier: Apache-2.0
+*/}}
+
+{{- if and (.Capabilities.APIVersions.Has "monitoring.coreos.com/v1") .Values.thanos.ruler.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ThanosRuler
+metadata:
+  name: {{ include "release.name" . }} # ressources name = thanos-ruler-<name>
+  labels:
+    {{- include "plugin.labels" . | nindent 4 }}
+    {{- include "thanos.labels" . | nindent 4 }}
+    {{- if .Values.thanos.ruler.labels }}
+    {{ toYaml .Values.thanos.ruler.labels | nindent 4 }}
+    {{- end }}
+spec:
+  image: "{{ .Values.thanos.image.repository }}:{{ .Values.thanos.image.tag | default .Chart.AppVersion }}"
+  ruleSelector:
+    matchLabels:
+      thanos-ruler: {{ .Values.thanos.ruler.matchLabel | default .Release.Name }} # Label for fetching rules
```

```suggestion
      thanos-ruler: {{ .Values.thanos.ruler.matchLabel | default .Release.Name }}
```

either the comment goes above or we leave it for good. decide if you still want do have it in for understanding purposes.